.github/workflows/terraform-observe_release.yaml

```diff
@@ -16,7 +16,20 @@ on:
         default: '0'
 
 jobs:
+  check-branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.get_branch.outputs.branch }}
+    steps:
+      - name: get_branch
+        id: get_branch
+        run: |
+          echo "branch=$(echo ${GITHUB_REF##*/})" >> $GITHUB_OUTPUT
+          echo "will only release if branch is main: ${GITHUB_REF##*/}"
+
   bump:
+    needs: [check-branch]
+    if: needs.check-branch.outputs.branch == 'main'
```

Comment:
@slobsv Will this always be main or a version of `refs/heads/<branch_name>` since you're using `GITHUB_REF` notation

Reply:
It's being chopped off using shell:

```sh
${GITHUB_REF##*/}
```

Which FWIW is why you should generally consider the env vars legacy and avoid them if at all possible:

https://github.com/observeinc/.github/pull/86/files#r1290435023

You can only have so many (flat) env vars before it becomes impossible to find any. Whereas the `github` context among others is an object and can be organized into nested sub-objects. So generally there's far less need to parse context values than environment variables to get the result you want. And parsing strings in shell tends to be really hard to read.

If you were stuck using env, `endsWith` would be the most readable way to achieve that versus attempting to extract the suffix and compare.

https://docs.github.com/en/actions/learn-github-actions/expressions#endswith